src/ansys/fluent/core/launcher/container_launcher.py

```diff
@@ -192,54 +197,30 @@ def __init__(
         In job scheduler environments (e.g., SLURM, LSF, PBS), resources and compute nodes are allocated,
         and core counts are queried from these environments before being passed to Fluent.
         """
-        insecure_mode_env = os.getenv("PYFLUENT_CONTAINER_INSECURE_MODE") == "1"
-        if certificates_folder is None and not insecure_mode and not insecure_mode_env:
-            raise ValueError(
-                "To launch Fluent in secure gRPC mode, set `certificates_folder`."
-            )
-        if certificates_folder is not None and insecure_mode:
-            raise ValueError(
-                "`certificates_folder` and `insecure_mode` cannot be set at the same time."
-            )
-
-        locals_ = locals().copy()
-        argvals = {
-            arg: locals_.get(arg)
-            for arg in inspect.getargvalues(inspect.currentframe()).args
-        }
-        self.argvals, self.new_session = _get_argvals_and_session(argvals)
-        if self.argvals["start_timeout"] is None:
+        self.argvals, self.new_session = _get_argvals_and_session(
```

Comment:
This was changed as sadly there's no way to type transform a typedict to have optional keys (python isn't typescript, sad)